### ISSUE TITLE
warnings are returned by make when command sets warn=-1

### ISSUE THREAD

**Issue by johnbaums** [ropensci/drake#1049]
## Prework

- [x] Read and abide by `drake`'s [code of conduct](https://github.com/ropensci/drake/blob/master/CODE_OF_CONDUCT.md).
- [x] Search for duplicates among the [existing issues](https://github.com/ropensci/drake/issues), both open and closed.
- [x] Advanced users: verify that the bug still persists in the current development version (i.e. `remotes::install_github("ropensci/drake")`) and mention the [SHA-1 hash](https://git-scm.com/book/en/v1/Getting-Started-Git-Basics#Git-Has-Integrity) of the [Git commit you install](https://github.com/ropensci/drake/commits/master).

## Description

Warnings are returned when making a plan despite the warning being ignored by the target's command. E.g. the function `raster::setMinMax` sets `options(warn=-1)` so that a inevitable warning is ignored, but the handler used by drake doesn't respect `warn=-1`. Is this intentional?

## Reproducible example

``` r
library(raster)
#> Loading required package: sp
library(drake)

p <- drake_plan(
  x = {
    r <- raster(extent(0, 1, 0, 1), res=0.01)
    r[1] <- 1
    r
  }
)

make(p)
#> target x
#> Warning: target x warnings:
#>   no non-missing arguments to min; returning Inf
#>   no non-missing arguments to max; returning -Inf

## More generally...

library(drake)

f <- function() {
  w <- options(warn=-1)
  on.exit(options(w))
  warning('This warning is ignored.')
}

p <- drake_plan(
  x = f()
)

make(p)
#> target x
#> Warning: target x warnings:
#>   This warning is ignored.
```

<sup>Created on 2019-11-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

<details>

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.0 (2019-04-26)
#>  os       macOS Mojave 10.14.3        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Melbourne         
#>  date     2019-11-05                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                         
#>  assertthat    0.2.1      2019-03-21 [2] CRAN (R 3.6.0)                 
#>  backports     1.1.5      2019-10-02 [1] CRAN (R 3.6.0)                 
#>  base64url     1.4        2018-05-14 [2] CRAN (R 3.6.0)                 
#>  cli           1.1.0      2019-03-19 [2] CRAN (R 3.6.0)                 
#>  codetools     0.2-16     2018-12-24 [2] CRAN (R 3.6.0)                 
#>  crayon        1.3.4      2017-09-16 [2] CRAN (R 3.6.0)                 
#>  digest        0.6.21     2019-09-20 [1] CRAN (R 3.6.0)                 
#>  drake       * 7.7.0.9002 2019-11-05 [1] Github (ropensci/drake@b499a04)
#>  evaluate      0.14       2019-05-28 [2] CRAN (R 3.6.0)                 
#>  filelock      1.0.2      2018-10-05 [2] CRAN (R 3.6.0)                 
#>  highr         0.8        2019-03-20 [2] CRAN (R 3.6.0)                 
#>  htmltools     0.4.0      2019-10-04 [1] CRAN (R 3.6.0)                 
#>  igraph        1.2.4.1    2019-04-22 [2] CRAN (R 3.6.0)                 
#>  knitr         1.25       2019-09-18 [1] CRAN (R 3.6.0)                 
#>  lattice       0.20-38    2018-11-04 [2] CRAN (R 3.6.0)                 
#>  magrittr      1.5        2014-11-22 [2] CRAN (R 3.6.0)                 
#>  pillar        1.4.2      2019-06-29 [2] CRAN (R 3.6.0)                 
#>  pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 3.6.0)                 
#>  R6            2.4.0      2019-02-14 [2] CRAN (R 3.6.0)                 
#>  raster      * 3.0-7      2019-09-24 [1] CRAN (R 3.6.0)                 
#>  Rcpp          1.0.2      2019-07-25 [1] CRAN (R 3.6.0)                 
#>  rlang         0.4.0      2019-06-25 [1] CRAN (R 3.6.0)                 
#>  rmarkdown     1.16       2019-10-01 [1] CRAN (R 3.6.0)                 
#>  sessioninfo   1.1.1      2018-11-05 [2] CRAN (R 3.6.0)                 
#>  sp          * 1.3-1      2018-06-05 [2] CRAN (R 3.6.0)                 
#>  storr         1.2.1      2018-10-18 [2] CRAN (R 3.6.0)                 
#>  stringi       1.4.3      2019-03-12 [2] CRAN (R 3.6.0)                 
#>  stringr       1.4.0      2019-02-10 [2] CRAN (R 3.6.0)                 
#>  tibble        2.1.3      2019-06-06 [2] CRAN (R 3.6.0)                 
#>  txtq          0.2.0      2019-10-15 [1] CRAN (R 3.6.0)                 
#>  withr         2.1.2      2018-03-15 [2] CRAN (R 3.6.0)                 
#>  xfun          0.10       2019-10-01 [1] CRAN (R 3.6.0)                 
#>  yaml          2.2.0      2018-07-25 [2] CRAN (R 3.6.0)                 
#> 
#> [1] /Users/jbau/Library/R/3.6/library
#> [2] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

## Expected output

As above but without the warnings.

**Comment by wlandau** [comment]
By design, `drake` logs warnings so you can dig them up later with `diagnose(your_target)$warnings`.

https://github.com/ropensci/drake/blob/b499a04337027ea02ca93bf20794c51c2628a087/R/local_build.R#L174-L177

And it makes sure to tell you about any warnings as soon as the target finishes. The `warn` option is temporarily set to `>= 1` because `drake` is designed for long computations, so it is usually best to see warnings as soon as possible.

https://github.com/ropensci/drake/blob/b499a04337027ea02ca93bf20794c51c2628a087/R/local_build.R#L454-L463

In other words, `drake` overrides `warn = -1`, and it has good reasons. If you want to tone down the verbosity of warnings, you could try setting `warning.length`, which `drake` does not bother to modify.
